src/main/java/org/sigmah/offline/js/AuthenticationJS.java

```diff
@@ -155,34 +169,114 @@ public native void setOrganizationLogo(String organizationLogo) /*-{
 		this.organizationLogo = organizationLogo;
 	}-*/;
 
-	public Integer getOrgUnitId() {
+	public Integer getMainOrgUnitId() {
 		return Values.getInteger(this, "orgUnitId");
 	}
 
-	public void setOrgUnitId(Integer orgUnitId) {
+	public void setMainOrgUnitId(Integer orgUnitId) {
 		Values.setInteger(this, "orgUnitId", orgUnitId);
 	}
 
-	public native ProfileJS getAggregatedProfile() /*-{
-		return this.aggregatedProfile;
+	public Set<Integer> getSecondaryOrgUnitIdsDTO() {
+		Set<Integer> orgUnitIds = new HashSet<Integer>();
+		JsArrayInteger secondaryOrgUnitIds = getSecondaryOrgUnitIds();
+		if (secondaryOrgUnitIds == null) {
+			return Collections.emptySet();
```

Comment:
It's strange to declare and initialize `orgUnitIds` before returning `Collections.emptySet()`. Move `orgUnitIds` declaration after the if or return it line 184.